src/app/[locale]/events/(components)/EventsPage.tsx

```diff
@@ -282,13 +282,16 @@ function CustomHits({ isPastEvent }: { isPastEvent: boolean }) {
       let month = startDate.getMonth();
       let year = startDate.getFullYear();
       let key = `${year}-${month + 1}`;
+      if (isPastEvent && !hit.show_in_past_events) {
+        return;
```

Comment:
this filter should be added in `getEventFilter` function so we don't fetch unwanted entries

Reply:
done!